components/push/src/crypto.rs

```diff
@@ -9,132 +9,41 @@
 //! This uses prime256v1 EC encryption that should come from internal crypto calls. The "application-services"
 //! module compiles openssl, however, so might be enough to tie into that.
 
-use log;
-use std::clone;
-use std::cmp;
-use std::fmt;
-
+use crate::error;
 use ece::{
-    Aes128GcmEceWebPushImpl, AesGcmEceWebPushImpl, AesGcmEncryptedBlock, LocalKeyPair,
-    LocalKeyPairImpl,
+    Aes128GcmEceWebPushImpl, AesGcmEceWebPushImpl, AesGcmEncryptedBlock, EcKeyComponents,
+    LocalKeyPair, LocalKeyPairImpl,
 };
-use openssl::ec::EcKey;
+use log;
 use openssl::rand::rand_bytes;
-
-use crate::error;
+use serde_derive::*;
 
 pub const SER_AUTH_LENGTH: usize = 16;
 pub type Decrypted = Vec<u8>;
 
-/// build the key off of the OpenSSL key implementation.
-/// Much of this is taken from rust_ece/crypto/openssl/lib.rs
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
 pub struct Key {
-    /// A "Key" contains the cryptographic Web Push Key data.
-    private: LocalKeyPairImpl,
-    pub public: Vec<u8>,
+    ec_key: EcKeyComponents,
```

Comment:
Similar comment here re: naming of the field, not a big deal but something worth thinking about.

Reply:
This struct has lost the explicit version number that was on the previous serialization, and which feels like it could be useful to our future selves. What do you think about adding it back?  /cc @jrconlin

Reply:
I'd prefer a version number somewhere in it, to be honest. Versioning data like this has gotten me out of messes more often than not. I tend to say that you only version things you want to keep and improve.

Reply:
Added versions for both keys!